src/NServiceBus.Core/NServiceBus.Core.csproj

```diff
@@ -28,6 +28,7 @@
     <PackageReference Include="Fody" Version="6.2.4" PrivateAssets="All" />
     <PackageReference Include="Janitor.Fody" Version="1.8.0" PrivateAssets="All" />
     <PackageReference Include="LightInject.Source" Version="5.0.3" PrivateAssets="All" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="3.1.7" />
```

Comment:
what version should we target here? would it make sense to go for 3.1.0 or even 3.0.0 if possible?

Reply:
Shall we use dependency ranges here?

```
<PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="[3.1.0, 4.0.0)" />
```

This is the first time Core brings in an external dependency that might also affect/be affected by the user environment. Right?

Reply:
Imo we should probably not use a n upper range and afaik there has been a conversation about this somewhere with the recommendation to not use upper ranges for Microsoft packages like this? NServiceBus.Extensions.DependencyInjection doesn't define an upper range: https://github.com/Particular/NServiceBus.Extensions.DependencyInjection/blob/master/src/NServiceBus.Extensions.DependencyInjection/NServiceBus.Extensions.DependencyInjection.csproj#L14 nor does hosting: https://github.com/Particular/NServiceBus.Extensions.Hosting/blob/master/src/NServiceBus.Extensions.Hosting/NServiceBus.Extensions.Hosting.csproj#L14

Reply:
I might be missing something here.

By not setting a range does it mean that if each time 3.x minor or patch is released we need to release a new version of Core to allow customers to use that?

I remember we had to do it for each of the Newtonsoft releases until we decided to use a [dependency range](https://github.com/Particular/NServiceBus.Newtonsoft.Json/blob/master/src/NServiceBus.Newtonsoft.Json/NServiceBus.Newtonsoft.Json.csproj).

Reply:
That's part of the discussion

https://github.com/Particular/NServiceBus.Extensions.Hosting/pull/47#issuecomment-618672245

I can't find the rest though. I thought @bording was supposed to raise something somewhere but I can't remember.

If you look at NServiceBus.Extension.Hosting you can see that even without range it means `>= `

![image](https://user-images.githubusercontent.com/174258/90604497-3592a580-e1fd-11ea-86b2-d478cd855f0c.png)

Maybe you can consult with Brandon, write it down and then RFC it?